### PR TITLE
Improve Cursor Handling *FUBAR*

### PR DESCRIPTION
**Pull request by RobertBColton**
I was looking at the cursor function and tweaked it to follow the MSDN example of how to use it. Tested to confirm the cursor is still only applied to the client area as GM does and is recommended by MSDN.

1) WM_SETCURSOR should return TRUE when you set the cursor
2) WM_SETCURSOR did not need an else, just break for DefWindowProc
3) Removed lingering comments about DirectShow and touch extensions, if we decide to add them, then just add working code, there's plenty of documentation about how DShow works online, the comments serve no purpose
4) WM_SETCURSOR should not handle the LoadCursor call, that's inefficient
5) `cr_drag` is the hand cursor in GMS: https://docs.yoyogames.com/source/dadiospice/002_reference/windows%20and%20views/the%20game%20window/window_set_cursor.html
6) Use fall-through for cr_default/cr_arrow
7) Set the default of cursor_sprite to -1 as specified by the GM docs:
http://gamemaker.info/en/manual#cursor_sprite
https://docs.yoyogames.com/source/dadiospice/002_reference/miscellaneous/cursor_sprite.html
8) Made action_set_cursor change cursor_sprite which it should from testing GM8 and GMS. That way the latter can also override the former.
9) Added code to screen_redraw to actually draw the cursor_sprite (more reason to move screen_redraw to `General/`)

# Segmentation Fault
There is only one problem right now. As the GameMaker docs specify, there is no way to specify an image_index for cursor_sprite, but it will play its animation at room speed if the sprite has multiple subimages. The only way to achieve this is by passing -1 for image_index to our draw_sprite call in screen_redraw, but that leads to a segmentation fault because of #865.

The default of cursor_sprite is also -1, even in GMS, which causes a segmentation fault in draw_sprite in run mode.

### CODE REVIEW

**Comment by JoshDreamland** [comment]
Why are we drawing the cursor for the system? Why is `cursor_sprite` a thing?

**Comment by RobertBColton** [comment]
@JoshDreamland As you can see from 8, it is because that was added in GM8 to override the drag and drop action. Then you could do it through code easily without any drag and drop. It's just a global added in that version. So it's not just to make it easy by doing something for the user, but making it possible to override the drag and drop action (which was an action created by Mark Overmars and does exist in GM5).

Mouse Cursor GM5

**Comment by JoshDreamland** [comment]
Does Windows not allow setting the cursor to an arbitrary bitmap? Because Linux (X11) does.

**Comment by RobertBColton** [comment]
@JoshDreamland yes but that is extraordinarily more complicated to load the bitmap for Win32, especially every time the user sets the `cursor_sprite` global, and it is also GM incompatible. For example, it will not be drawn at mouse_x and mouse_y and thus is not effected by scaling.

File: ENIGMAsystem/SHELL/Graphics_Systems/OpenGL1/GLscreen.cpp [hunk 1]
```diff
@@ -305,6 +307,7 @@ void screen_redraw()
     // glReadPixels(0, 0, width, height, GL_BGRA, GL_UNSIGNED_BYTE, pixels);
   }
 
+  draw_sprite(cursor_sprite, 0, mouse_x, mouse_y);
```
Comment (JoshDreamland) [inline ENIGMAsystem/SHELL/Graphics_Systems/OpenGL1/GLscreen.cpp]: Did you test this? As far as I can tell, `draw_sprite` has no special handling for the `-1` case. This should cause all kinds of problems.

Comment (RobertBColton) [reply ENIGMAsystem/SHELL/Graphics_Systems/OpenGL1/GLscreen.cpp]: Yes and that's why I have "SEGMENTATION FAULT" in bold letters in the OP above. GameMaker defaults it to -1, I don't know what you want me to do, check if it's > 0 before calling draw_sprite?

**Comment by RobertBColton** [comment]
Closing as fubar, see #1240

**Pull request closed without merging**